### PR TITLE
enhance: message on checking NIK status

### PR DESCRIPTION
**Pull request by indraprasetya154**
## Overview
- enhance: message on checking NIK status

## Related PR
- https://gitlab.com/jdsteam/sikd-website/-/merge_requests/475

## Evidence
title: enhance: message on checking NIK status mobile
project: SIKD
participants: @indraprasetya154 @samudra-ajri @azophy @fajarhikmal214

### CODE REVIEW

File: src/app/GraphQL/Mutations/DraftSignatureMutator.php [hunk 1]
```diff
@@ -52,17 +53,21 @@ public function signature($rootValue, array $args)
...
+            $signature = $this->doSignature($setupConfig, $draft, $passphrase);
 
-        $draft->Konsep = DraftConceptStatusTypeEnum::SENT()->value;
-        $draft->save();
+            $draft->Konsep = DraftConceptStatusTypeEnum::SENT()->value;
+            $draft->save();
 
-        return $signature;
+            $logData['status'] = KafkaStatusTypeEnum::SUCCESS();
+            $this->kafkaPublish('analytic_event', $logData);
```
Comment (samudra-ajri) [inline src/app/GraphQL/Mutations/DraftSignatureMutator.php:65]: For the suggestion.
Would you like to add the `log event` for this action?
For example `$logData['event']`

Or, if there is no initial object, you can call this func directly. For instance:
```
$this->kafkaPublish('analytic_event', [
            'event' => 'BSRE signed',
            'status' => KafkaStatusTypeEnum::SUCCESS(),
        ]);
```

This `event` key will be filtered the action on the ellastic consumer.

File: src/app/Http/Traits/SignatureTrait.php [hunk 2]
```diff
@@ -53,6 +57,34 @@ public function checkUserSignature($setupConfig)
...
+            $logData['longMessage'] = 'User NIK Anda Belum Terdaftar';
+        }
+
+        if ($checkUserResponse->status_code == BsreStatusTypeEnum::RESPONSE_CODE_BSRE_ACCOUNT_ESIGN_NOT_ACTIVE()->value) {
+            $logData['message'] = 'Sertifikat Tidak Aktif';
+            $logData['longMessage'] = 'User NIK sudah terdaftar tapi belum memiliki sertifikat esign yang aktif';
+        }
+
+        $this->kafkaPublish('analytic_event', $logData);
+        throw new CustomException($logData['message'], $logData['longMessage']);
```
Comment (samudra-ajri) [inline src/app/Http/Traits/SignatureTrait.php:85]: For sugesstion.
This publisher is also need the `event` key as the log. 😁

**Review by samudra-ajri — APPROVED** [review]

**Review by samudra-ajri — COMMENTED** [review]
TGIF! 🌻

File: src/app/Enums/BsreStatusTypeEnum.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * @method static self RESPONSE_CODE_BSRE_ACCOUNT_NOT_REGISTERED()
+ * @method static self RESPONSE_CODE_BSRE_ACCOUNT_ESIGN_NOT_ACTIVE()
+ */
+
+class BsreStatusTypeEnum extends Enum
+{
+    protected static function values(): array
+    {
+        return [
+            'RESPONSE_CODE_BSRE_ACCOUNT_OK' => 1111,
```
Comment (fajarhikmal214) [inline src/app/Enums/BsreStatusTypeEnum.php:18]: wanna ask, is there no other possibilities ? or not 'cause there is no documentation ?

**Review by azophy — APPROVED** [review]

File: src/app/Enums/BsreStatusTypeEnum.php (hunk 3 above)
Comment (indraprasetya154) [reply src/app/Enums/BsreStatusTypeEnum.php:18]: In the past, I've been asked for documentation status code, but sadly I didn't get it until now, so I create this enum base on my exploration at swagger 🐱

**Review by fajarhikmal214 — APPROVED** [review]

File: src/app/Http/Traits/SignatureTrait.php [hunk 4]
```diff
@@ -53,6 +57,34 @@ public function checkUserSignature($setupConfig)
...
+     * @param  mixed $checkUserResponse
+     * @return void
+     */
+    public function invalidResponseCheckUserSignature($checkUserResponse)
+    {
+        $logData = [
+            'message' => 'Invalid BSRE Service',
+            'longMessage' => 'Tidak dapat terhubung dengan BSRE, silahkan coba kembali',
+            'serviceResponse' => (array) $checkUserResponse
+        ];
```
[the remark this answers is not in the document]
Comment (indraprasetya154) [reply src/app/Http/Traits/SignatureTrait.php]: Fixed here [af5f758](https://github.com/jabardigitalservice/office-services/pull/493/commits/af5f7580f230f018863d506ebcf552d45b50edbe)

File: src/app/Http/Traits/SignatureTrait.php (hunk 2 above)
Comment (indraprasetya154) [reply src/app/Http/Traits/SignatureTrait.php:86]: Fixed here [af5f758](https://github.com/jabardigitalservice/office-services/pull/493/commits/af5f7580f230f018863d506ebcf552d45b50edbe)

File: src/app/GraphQL/Mutations/DraftSignatureMutator.php (hunk 1 above)
Comment (indraprasetya154) [reply src/app/GraphQL/Mutations/DraftSignatureMutator.php:65]: Fixed here [af5f758](https://github.com/jabardigitalservice/office-services/pull/493/commits/af5f7580f230f018863d506ebcf552d45b50edbe)

**Review by samudra-ajri — APPROVED** [review]

**Pull request merged**